microservices/site-launch/lambda-functions/general-domain-validation/index.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  AmplifyClient,
+  GetDomainAssociationCommand,
+  DomainStatus,
+} from "@aws-sdk/client-amplify"
+import type {
+  GetDomainAssociationCommandInput,
+  GetDomainAssociationCommandOutput,
+} from "@aws-sdk/client-amplify"
+
+import logger from "../../shared/logger"
+import {
+  SITE_LAUNCH_LAMBDA_TYPE,
+  SITE_LAUNCH_LAMBDA_STATUS,
+} from "../../shared/types" // TODO: Change to aliased imports
+
+interface GeneralDomainValidationLambdaParams {
+  appId: string
+  primaryDomain: string
+}
+
+interface GeneralDomainValidationLambdaResponse {
+  lambdaType: SITE_LAUNCH_LAMBDA_TYPE
+  status: SITE_LAUNCH_LAMBDA_STATUS
+  appId: string
+  primaryDomain: string
+}
+
+const SUCCESSFUL_GENERAL_DOMAIN_VALIDATION_STATUSES = [
+  DomainStatus.AVAILABLE,
+  DomainStatus.PENDING_DEPLOYMENT,
+]
+
+export const generalDomainValidation = async (
+  event: GeneralDomainValidationLambdaParams
+): Promise<GeneralDomainValidationLambdaResponse> => {
+  logger.info(event)
+
+  const { AWS_REGION } = process.env
+  const amplifyClient = new AmplifyClient({
+    region: AWS_REGION,
+  })
+
+  // Validation check
+  const { appId, primaryDomain } = event
+
+  if (!appId) throw new Error(`appId was undefined`)
+  if (!primaryDomain) throw new Error(`primaryDomain was undefined`)
+
+  const params: GetDomainAssociationCommandInput = {
+    appId,
+    domainName: primaryDomain,
+  }
+  const getDomainAssociationCommand = new GetDomainAssociationCommand(params)
+
+  try {
```

Comment:
I see this as a convenient way to log all errors. Eventually, we would still have to throw the errors to trigger the failure lambda. Do you see a better way to do this?

Reply:
from the docs [here](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html) under `Error output`, it seems that the output would already be transformed to `$.Cause` if nothing is specified. 

since we match on `States.ALL`, we can just log directly in our failure notification lambda rather than having to do it 3 separate times. this also allows us to omit the `try/catch` block

Reply:
This is fair - f85fdb6